Add torch compile force disable caches alias

oulgen opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #158072

Bunch of people keep thinking current alias only disables inductor cache because it has the name inductor in it. lets globalize the name

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @Lucaskabela

oulgen committed:
Add torch compile force disable caches alias

[ghstack-poisoned]

ezyang commented:
May I propose we also move this config to torch.compiler.config then? You can keep an alias to the new location from the old one.

oulgen committed:
None on "Add torch compile force disable caches alias"


Bunch of people keep thinking current alias only disables inductor cache because it has the name inductor in it. lets globalize the name

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

oulgen committed:
None on "Add torch compile force disable caches alias"


Bunch of people keep thinking current alias only disables inductor cache because it has the name inductor in it. lets globalize the name

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

oulgen committed:
Update

[ghstack-poisoned]

oulgen committed:
Update

[ghstack-poisoned]

jeffdaily commented:
Failure info:

https://hud.pytorch.org/failure?name=rocm-mi300%20%2F%20linux-noble-rocm-py3.12-mi300%20%2F%20test%20(default%2C%203%2C%206%2C%20linux.rocm.gpu.mi300.2%2C%20unstable)&jobName=linux-noble-rocm-py3.12-mi300%20%2F%20test%20(default%2C%203%2C%206%2C%20linux.rocm.gpu.mi300.2%2C%20unstable)&failureCaptures=inductor%2Ftest_inductor_scheduler.py%3A%3ATestSchedulerCUDA%3A%3Atest_flop_counter_op_options0_cuda_float32

oulgen committed:
Update on "Add torch compile force disable caches alias"


Bunch of people keep thinking current alias only disables inductor cache because it has the name inductor in it. lets globalize the name

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben Lucaskabela

[ghstack-poisoned]

oulgen committed:
Update on "Add torch compile force disable caches alias"


Bunch of people keep thinking current alias only disables inductor cache because it has the name inductor in it. lets globalize the name

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben Lucaskabela

[ghstack-poisoned]

oulgen commented:
@pytorchbot merge
